external/fv3fit/fv3fit/reservoir/config.py

```diff
@@ -59,47 +59,46 @@ class BatchLinearRegressorHyperparameters:
 
 
 @dataclass
-class ReservoirTrainingConfig:
+class ReservoirTrainingConfig(Hyperparameters):
     """
+    input_variables: variables and additional features in time series
+    output_variables: time series variables, must be subset of input_variables
     reservoir_hyperparameters: hyperparameters for reservoir
     readout_hyperparameters: hyperparameters for readout
-    n_burn: number of training samples to discard from beginning of training
-        time series.
+    n_batches_burn: number of training batches at start of time series to use
+        for synchronizaton.  This data is  used to update the reservoir state
+        but is not included in training.
     input_noise: stddev of normal distribution which is sampled to add input
         noise to the training inputs when generating hidden states. This is
         commonly done to aid in the stability of the RC model.
     seed: random seed for sampling
-    n_samples: number of samples to use in training
-    subdomain: Optional subdomain config. If provided, one reservoir and readout
-        are created and trained for each subdomain. Subdomain size and reservoir
-        input size much match.
+    subdomain: Subdomain config. All subdomains use the same reservoir weights;
+        one readout is created and trained for each subdomain. Subdomain size
+        and reservoir input size much match.
     square_half_hidden_state: if True, square even terms in the reservoir
         state before it is used as input to the regressor's .fit and
         .predict methods. This option was found to be important for skillful
         predictions in Wikner+2020 (https://doi.org/10.1063/5.0005541)
+    autoencoder_path: optional path for autoencoder to use in encoding time series
+        before passing to reservoir
     """
 
+    input_variables: Sequence[str]
+    output_variables: Sequence[str]
     subdomain: CubedsphereSubdomainConfig
     reservoir_hyperparameters: ReservoirHyperparameters
     readout_hyperparameters: BatchLinearRegressorHyperparameters
-    n_burn: int
+    n_batches_burn: int
     input_noise: float
-    timestep: float
     seed: int = 0
-    n_samples: Optional[int] = None
     n_jobs: Optional[int] = -1
     square_half_hidden_state: bool = False
-
+    autoencoder_path: Optional[str] = None
     _METADATA_NAME = "reservoir_training_config.yaml"
 
-    def __post_init__(self):
-        if self.subdomain is not None:
-            if (
-                self.subdomain.size + 2 * self.subdomain.overlap
-            ) != self.reservoir_hyperparameters.input_size:
-                raise ValueError(
-                    "Subdomain size + overlaps and reservoir input_size must match."
-                )
+    @property
+    def variables(self) -> Set[str]:
+        return set(self.input_variables).union(self.output_variables)
```

Comment:
Is this necessary if `output_variables` is a subset of `input_variables`, as stated in the docstring?

Reply:
No, updated